src/main/java/hudson/plugins/translation/ResourceBundleFactoryImpl.java

```diff
@@ -30,20 +30,31 @@ protected Properties wrapUp(String locale, Properties props) {
             protected Properties get(String key) {
                 int mc = reloadModCount;
                 if (modCount!=mc) {
-                    clearCache();
+                    ResourceBundleFactoryImpl.this.clearCache();
                     modCount = mc;
                 }
                 return super.get(key);
             }
+
+            @Override
+            public boolean equals(Object o) {
+                // Modifications count should not been taken into account in equals() 
+                return super.equals(o);
+            }
+            
+            @Override
+            public int hashCode() {
+                return super.hashCode();
+            }       
         };
     }
 
     /**
      * Used to force the reloading of a cache.
      */
-    private volatile int reloadModCount = 0;
+    private int reloadModCount = 0;
```

Comment:
Might be better to change this to an `AtomicInteger` as that requires a lesser synchronization for read access than a monitor lock.